Sming/SmingCore/Platform/System.cpp

```diff
@@ -40,7 +40,7 @@ void SystemClass::initialize()
 	// Initialise the global task queue
 	system_os_task(taskHandler, USER_TASK_PRIO_1, taskQueue, TASK_QUEUE_LENGTH);
 
-	system_init_done_cb(staticReadyHandler);
+	system_init_done_cb([]() { System.readyHandler(); });
```

Comment:
That sounds like a bad design to me. You have a class that relies on an instance of that class to exist and to be named System. It will work but is prone to brake.

Reply:
It replaces the previous static method so they're equivalent. We need some way to get `this` to the callback, so if we don't use `System` then it would be another global static.

```
void SystemClass::staticReadyHandler()
{
	System.readyHandler();
}
```

Reply:
As there's only one `SystemClass` instance we could redefine all the mamber variables as `static` ?